python/ray/data/read_api.py

```diff
@@ -1235,6 +1248,37 @@ def read_binary_files(
     )
 
 
+@PublicAPI(stability="alpha")
+def read_sql(
+    sql: str,
+    connection_factory: Callable[[], Connection],
+    *,
+    parallelism: int = -1,
+    ray_remote_args: Optional[Dict[str, Any]] = None,
+):
+    """Read from a database that provides a
+    `Python DB API2-compliant <https://peps.python.org/pep-0249/>`_ connector.
+
+    Args:
+        sql: The SQL query to execute.
+        connection_factory: A function that takes no arguments and returns a
+            Python DB API2
+            `Connection object <https://peps.python.org/pep-0249/#connection-objects>`_.
+        parallelism: The requested parallelism of the read.
+        ray_remote_args: Keyword arguments passed to :func:`ray.remote` in read tasks.
+
```

Comment:
Could you also add some code examples here? Some users might directly go to our API reference page here.